index.js

```diff
@@ -1046,14 +1077,14 @@ function toHex (buf) {
   return buf && b4a.toString(buf, 'hex')
 }
 
-function preappend (blocks) {
+async function preappend (blocks) {
   const offset = this.state.length
   const fork = this.state.encryptionFork
 
   if (this.encryption.compat !== this.core.compat) this._updateEncryption()
 
   for (let i = 0; i < blocks.length; i++) {
```

Comment:
Should this maybe be done in parallel using something like `await Promise.all(blocks.map(encrypt))`? This seems like a major perf regression otherwise.

Reply:
same as today since sync right now, but we can adjust later. the async'ness is there for flexibility. for now lets keep it simple.